### PR TITLE
Add `Target` and `Generator.processTargetMerges()`

### PR DESCRIPTION
**Pull request by brentleyjones**

### CODE REVIEW

File: tools/generator/src/BuildSetting.swift [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    /// becomes an array of `String`s).
+    var asAny: Any {
+        switch self {
+        case .string(let string):
+            return string
+        case .bool(let bool):
+            return bool ? "YES" : "NO"
+        case .array(let array):
+            if array.count < 2 {
+                return array.first ?? ""
```
Comment (tylerwilliams) [inline tools/generator/src/BuildSetting.swift:24]: ok, i'm dumb, can you explain this to me?

**Review by tylerwilliams — APPROVED** [review]
LGTM

File: tools/generator/src/BuildSetting.swift (hunk 1 above)
Comment (brentleyjones) [reply tools/generator/src/BuildSetting.swift:24]: If an array has 1 or 0 elements, Xcode wants it to be the same as a string.

Comment (brentleyjones) [reply tools/generator/src/BuildSetting.swift:24]: I'll minimally add a comment.

**Pull request merged**